### ISSUE TITLE
Exception: ZdcPluginMissing: SSL/TLS Plugin (Windows Pharo 9.0 32-bit)

### ISSUE THREAD

**Issue by PowellDean** [guillep/pharo-core#9730]
**Bug description**
Pharo 9.0 32-bit environment, launched from PharoLoader. This is a brand new image loaded 2021-08-04 (00:41 North America/Mountain timezone) directly through the Pharo Launcher:

Pharo9.0.0
build: 1542, commit: 7864e16, date: 1 August 2021 (per the Welcome screen)

Cannot update #appearance#current.
Exception: ZdcPluginMissing: SSL/TLS Plugin
Initialization failed (VM plugin missing ? OS libraries missing)

**To Reproduce**
1. Open the PharoLauncher on Windows 10
2. Choose a new Pharo 9.0 32-bit image
3. Error appears when I launch the image

Please note: I am unable to determine if this same error occurs with the 9.0 64-bit image, as that image appears to be completely broken with my version of Windows 10.

**Expected behavior**
This plugin is working fine through the Pharo 8.0 Old Stable image, both 32-bit and 64-bit on Windows. This error also was not occurring yesterday, as I was able to successfully make an API call to an HTTPS endpoint August 3. I had an existing Pharo 9.0 32-bit image for which this API call _was_ working. But upon returning to coding, the same error started occurring in that image. I spun up a rand new image to confirm the error is not with my code but with the image.

**Screenshots**
N/A

**Version information:**
 - OS: Windows 10 Enterprise
 - Version: Version 1909 (OS Build 18363.1679)
 - Pharo Version 9 32-bit

**Expected development cost**
I am willing to help if at all possible. Just ask via email.

**Additional context**

**Comment by Ducasse** [comment]
Hi Dean
Thanks a lot for your super nice report. We will have a look.

S

**Comment by Ducasse** [comment]
BTW why are you using a Pharo 32bits? I thought that windows 10 was 64 bits.

**Comment by PowellDean** [comment]
Hi  Stéphane:

Thanks for the followup. The Pharo 9 64-bit images have been very problematic for me on Windows 10 (Enterprise version). As of last night when I was testing, the 9.0 64-bit image on Windows was non-functional. I can successfully create the image, but when I open it it simply hangs. I will test again today on a different machine running a different edition of Windows 10 and let you know how that works.

I should open up another issue for the problems running Pharo 9 64-bit on Windows 10 Enterprise. Stay tuned for that.

Thanks!

Dean

**Comment by astares** [comment]
Did you try to update the VM in Pharo Launcher?

**Comment by Ducasse** [comment]
This is super important for us to know when Pharo does not work on a platform. 
This is strange that it does not work on Windows 10 because we tested it on our ARM tactile Windows machine. 
So please give us more information so that we can fix this.

**Comment by PowellDean** [comment]
@astares: Thank you for the suggestion. I had not tried that and can now confirm that the Pharo 9 32-bit image no longer throws the error. Thank you.

**Comment by PowellDean** [comment]
There are other issues brought up in this thread re: the Pharo 9 64-bit image as run on Windows 10. To avoid conflating them, I will post information here, then close this issue. Then I'll open a separate issue for tracking purposes, and I'll re-post my findings in that issue.

I have installed the latest PharoLauncher product on two separate Windows 10 machines:
* __Machine 1__: Windows 10 Enterprise Edition - Version 1909 (OS Build 18363.1679)
* __Machine 2__: Windows 10 Home Edition - Version 2004 (OS Build 19041.1052)

On BOTH machines I installed the current version of the PharoLauncher, as downloaded from the pharo.org website. I then installed the XML parsing libraries (XMLDOMParser and friends). I then installed the package I've been working on via Monticello.

On Machine 2: No issues at all running either the Pharo 9 32-bit or Pharo 9 64-bit image. External packages both loaded without issue, and my package was able to connect to an HTTPS: endpoint without any problem. Again, this is the case for BOTH 32- and 64-bit Pharo 9 images. The Windows 10 Home Edition experienced no issues.

On Machine 1:
Upon first launching the 64-bit image (via PharoLauncher), the main window appeared, but the Pharo image was completely unresponsive for ~30 seconds. After that time, the following error message popped up:

Information
cannot update #appearance#current
Exception: ConnectionTimeout: Cannot connect to 140.82.113.3:443<br>

After this, the image was responsive again and I was able to load both the XML parsing libraries (XMLDOMParser and friends) and my package through Monticello.

Upon running my package code, the system again becaome completely unresponsive for ~25-30 seconds, after which a walkback window appeared outlining the following:

* Class Socket
* Method: connectTo: hostAddress port: port waitForConnectionFor: timeout
* Error message: signal: ConnectionTimedOut. Cannot connect to xx.xx.xx.xx:443

On a whim, I launched the PharoLauncher "as Administrator", but I encountered the same results.

I should point out that the Windows 10 Enterprise Edition is my company laptop and is subject to company-installed Group Policy settings. However, I can confirm that the Pharo 8.0 Old Stable images (both 32- and 64-bit) run perfectly on this laptop. The code I wrote connects tothe remote HTTPS API endpoint.  I have full Local Admin privileges on this laptop.

**Comment by Ducasse** [comment]
Hello 
why did you close the issue?
Do you have a firewall?
Because it looks like pharo tried to do a request.

S

**Comment by PowellDean** [comment]
Hello:

The issue, as raised, is no longer happening. My Pharo 9 32-bit image is now working as expected, with no errors thrown when attempting to connect to the https: endpoint.

There is a standard Windows firewall on the machine, but I'm a Local Admin and I have full control over the rules. I can confirm I allowed Pharo to connect through the firewall. During the time the exception was being thrown, I checked the same code inside a Pharo 8 image (both 32- and 64-bit), and experienced no issue. So the firewall couldn't have been the cause. It was definitely a problem confined to the Pharo 9 image.

I outlined a number of other issues with the Pharo 9 64-bit image that are still ongoing. I didn't want to confuse the (mysteriously resolved) ZdcPluginMissing issue with the other issues, so I thought it best to close this one and open a separate one which outlines the wider issues with the Pharo 9 64-bit image on Windows Enterprise.

I apologize for any confusion. I'm happy to re-open this issue if it still warrants attention.

Regards,
Dean
